service/src/main/java/bio/terra/workspace/service/policy/flight/MergeBillingProfilePolicyAttributesStep.java

```diff
@@ -0,0 +1,101 @@
+package bio.terra.workspace.service.policy.flight;
+
+import bio.terra.policy.model.TpsComponent;
+import bio.terra.policy.model.TpsObjectType;
+import bio.terra.policy.model.TpsPaoConflict;
+import bio.terra.policy.model.TpsPaoGetResult;
+import bio.terra.policy.model.TpsPaoUpdateResult;
+import bio.terra.policy.model.TpsPolicyInputs;
+import bio.terra.policy.model.TpsUpdateMode;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.exception.InternalLogicException;
+import bio.terra.workspace.service.policy.TpsApiDispatch;
+import bio.terra.workspace.service.resource.exception.PolicyConflictException;
+import bio.terra.workspace.service.spendprofile.SpendProfileId;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import java.util.List;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MergeBillingProfilePolicyAttributesStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(MergeBillingProfilePolicyAttributesStep.class);
+
+  private final UUID workspaceId;
+  private final SpendProfileId spendProfileId;
+  private final TpsApiDispatch tpsApiDispatch;
+
+  public MergeBillingProfilePolicyAttributesStep(
+      UUID workspaceId, SpendProfileId spendProfileId, TpsApiDispatch tpsApiDispatch) {
+    this.workspaceId = workspaceId;
+    this.spendProfileId = spendProfileId;
+    this.tpsApiDispatch = tpsApiDispatch;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    // Create PAOs if they don't exist; catch TPS exceptions and retry
+    TpsPaoGetResult destinationPao;
+    UUID spendProfileUUID = UUID.fromString(spendProfileId.getId());
```

Comment:
This is running afoul of the fact that our test spend profile ID is not a UUID (wm-default-spend-profile).